frontend/src/metabase-lib/comparison.ts

```diff
@@ -8,3 +9,17 @@ export function areLegacyQueriesEqual(
 ): boolean {
   return ML.query_EQ_(query1, query2, fieldIds);
 }
+
+export function findColumnIndexForColumnSetting(
+  query: Query,
+  stageIndex: number,
+  columns: ColumnMetadata[],
```

Comment:
```suggestion
  columns: (ColumnMetadata | DatasetColumn)[],
```

Reply:
In fact, it seems I implemented this suggestion...